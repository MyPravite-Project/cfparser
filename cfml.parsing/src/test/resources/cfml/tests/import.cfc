import my.path.SomeComponent;

/**
* Hello world.
*/
component extends="my.path.BaseComponent" accessors="true"
{

}