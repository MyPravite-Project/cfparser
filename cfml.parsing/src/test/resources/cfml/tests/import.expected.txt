/*===TOKENS===*/
IMPORT                   <import>
IDENTIFIER               <my>
'.'                      <.>
IDENTIFIER               <path>
'.'                      <.>
IDENTIFIER               <SomeComponent>
';'                      <;>
Hidden:NEWLINE                  <>
Hidden:ML_COMMENT               </**
* Hello world.
*/>
Hidden:NEWLINE                  <>
COMPONENT                <component>
IDENTIFIER               <extends>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <my.path.BaseComponent>
CLOSE_STRING             <">
IDENTIFIER               <accessors>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <true>
CLOSE_STRING             <">
Hidden:NEWLINE                  <>
'{'                      <{>
Hidden:NEWLINE                  <>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (importStatement
    import
    (componentPath
      (multipartIdentifier
        (identifier my)
        .
        (identifierOrReservedWord (identifier path))
        .
        (identifierOrReservedWord (identifier SomeComponent))
      )
    )
    (endOfStatement ;)
  )
  (componentDeclaration
    component
    (componentAttribute
      (identifier extends)
      =
      (startExpression
        (compareExpression
          (baseExpression
            (unaryExpression
              (primaryExpression
                (literalExpression
                  (stringLiteral " (stringLiteralPart my.path.BaseComponent) ")
                )
              )
            )
          )
        )
      )
    )
    (componentAttribute
      (identifier accessors)
      =
      (startExpression
        (compareExpression
          (baseExpression
            (unaryExpression
              (primaryExpression
                (literalExpression (stringLiteral " (stringLiteralPart true) "))
              )
            )
          )
        )
      )
    )
    (componentGuts { })
  )
)
/*======*/
/*===DECOMPILE===*/
{
import my.path.SomeComponent;
  component extends='my.path.BaseComponent' accessors='true' {
};

}
/*======*/